tests/v2/validation/provisioning/permutations/permutations.go

```diff
@@ -55,14 +71,32 @@ func RunTestPermutations(s *suite.Suite, testNamePrefix string, client *rancher.
 				testClusterConfig.CNI = cni
 				name = testNamePrefix + " Node Provider: " + nodeProviderName + " Kubernetes version: " + kubeVersion + " cni: " + cni
 				s.Run(name, func() {
+					if testClusterConfig.CloudProvider == provisioninginput.AWSProviderName.String() {
```

Comment:
https://github.com/rancher/rancher/blob/54b0504e021f2cc42769392b79cc8c1012ab3210/tests/v2/validation/provisioning/permutations/permutations.go#L66

I've got a question here, in this triple loop we already have the provider info for that single provisioning action already. Why do we need to add another provider field to the config?

Reply:
there's a distinction between cloud provider and node provider that cross in times like these due to their close naming. In our case, we don't yet have a cloud provider option available in the go framework, which enable different settings on top of the node provider. Does this make more sense?